src/BuildScriptGenerator/Node/VersionProviders/NodeOnDiskVersionProvider.cs

```diff
@@ -3,13 +3,24 @@
 // Licensed under the MIT license.
 // --------------------------------------------------------------------------------------------
 
+using Microsoft.Extensions.Logging;
+
 namespace Microsoft.Oryx.BuildScriptGenerator.Node
 {
     public class NodeOnDiskVersionProvider : INodeVersionProvider
     {
+        private readonly ILogger<NodeOnDiskVersionProvider> _logger;
+
+        public NodeOnDiskVersionProvider(ILogger<NodeOnDiskVersionProvider> logger)
+        {
+            _logger = logger;
+        }
+
         // To enable unit testing
         public virtual PlatformVersionInfo GetVersionInfo()
         {
+            _logger.LogDebug("Getting list of versions from {installDir}", NodeConstants.InstalledNodeVersionsDir);
+
             var installedVersions = VersionProviderHelper.GetVersionsFromDirectory(
                         NodeConstants.InstalledNodeVersionsDir);
 
```

Comment:
should we also log the versions from installdir?

Reply:
I do not feel too strongly about it. We already support the more useful scenario of printing out list of supported versions when a user sees that a version they specified is not working.